hazelcast/future.py

```diff
@@ -58,10 +61,7 @@ def result(self):
         self._event.wait()
         if self._exception:
             re_raise(self._exception, self._traceback)
-        if self._result == NONE_RESULT:
-            return None
```

Comment:
No, actually this is just a simplification. Before that, we were holding `None` as the result initially. And, we have to have extra logic to support setting future's result to `None`. 

I made the initial result a unique object and simplified the code a bit. Now, when you set anything as the future's result (including `None`) it will be different than `_SENTINEL` and we will understand that the result is set.